feat(AAS-6): implement canvas and its annotation tools

andSoHenceforth opened the pull request:
## Description
Implemented the annotation canvas, which is where the user can make annotations to images, and implement relevant annotation tools that the user can use. 

While on the canvas, users can use the following tools:

- **Create bounding box tool:** Creates a bounding box that encapsulates an area of interest in the image
- **Select and move tool:** Selects an existing bounding box and then grants ability to move or resize it
- **Label tool:** Appends a text box to an existing bounding box as a descriptive label for the encapsulated area

<img width="1951" height="1412" alt="image" src="https://github.com/user-attachments/assets/c168904b-f0f1-42d2-8594-9b68478b9018" />
 
[Documentation of Canvas](https://avian-annotator.atlassian.net/wiki/x/BAAIAg)

## Required Checklist
[Jira Ticket Link](https://avian-annotator.atlassian.net/browse/AAS-6)

- [x] I have tested the code
- [x] I have updated documentation if needed
- [x] Jira ticket has been linked
- [ ] I have updated /.github/CODEOWNERS as appropriate (trial/testing)

## Optional Checklist
[Loom Video Link](https://www.loom.com/share/264414efbdac4a1c9f3b2f5cfb830a19?sid=1944fd29-6fb6-45ff-b360-0bd91c2ba54f)
- [ ] Tests have been added/updated as needed
- [x] I have linked a loom video

See [docs](https://avian-annotator.atlassian.net/wiki/x/AgD6) for how to fill this out and the motivation for this

V01DBREAKER committed:
feat(AA-29): add konva package

V01DBREAKER committed:
refactor(AA-29): make bounding box use konva

V01DBREAKER committed:
feat(AA-29): add tool changer, and wip selectmovetool

V01DBREAKER committed:
feat(AA-29): Add mock image to canvas

V01DBREAKER committed:
fix(AA-29): Make tool changer use index based tool change

V01DBREAKER committed:
feat(AA-29): Basic draggable objects

V01DBREAKER committed:
fix(AA-29): create independent boundingbox object and update x,y when dragged

V01DBREAKER committed:
feat(AA-29): complete select, transform and move tool

andSoHenceforth committed:
feat(AAS-24): implement label tool for annotations (#40)

* feat(AAS-24): implement label text tool

* feat(AAS-24): integrate label tool for annotations

windneverstops reviewed (approved):
LGTM.
Code quality looks fantastic.
Did not test it, but if it looks good for you locally, I'm happy

windneverstops commented on frontend/src/components/canvas/Tools/TemplateTool.tsx:
Assuming this is a template for a tool, might be worth renaming to `ToolTemplate` as I thought this was a literal tool which was called template when reading the file name

V01DBREAKER replied:
Resolved in [79c11ff43ead238a454009049f7799dbd302697a](https://github.com/avian-annotator/FYP/pull/41/commits/79c11ff43ead238a454009049f7799dbd302697a)

V01DBREAKER committed:
fix(AA-29): Make text always synced with box,

remove duplicate stageElements state,
prevent dragging of text,
prevent dragging of boxes when not selected

V01DBREAKER committed:
fix(AA-29): All errors ;-;

V01DBREAKER committed:
fix(AA-29): Rename to avoid confusion

V01DBREAKER commented:
NOTE:
<img width="390" height="42" alt="image" src="https://github.com/user-attachments/assets/0c716ba3-45f2-4895-903d-28a7715c4baa" />
I got sick of fixing errors. It works and passes all visual tests etc. The only major issue i have ignored is this:
https://github.com/avian-annotator/FYP/blob/13c622417d33d46005124ef8f3293c0786b443b7/frontend/src/components/canvas/Canvas.tsx#L9C1-L20C1
The types are kinda complicated due to the interchangable functions. When i have more time i'll fix this tech debt but its honestly just for code quality not for funcitonality because it works.

windneverstops commented:
> NOTE: <img alt="image" width="390" height="42" src="https://private-user-images.githubusercontent.com/29811739/484575446-0c716ba3-45f2-4895-903d-28a7715c4baa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY4MTkwNTEsIm5iZiI6MTc1NjgxODc1MSwicGF0aCI6Ii8yOTgxMTczOS80ODQ1NzU0NDYtMGM3MTZiYTMtNDVmMi00ODk1LTkwM2QtMjhhNzcxNWM0YmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTAyVDEzMTIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5NmRmMDU0Yjg3ZWNkZmJlNjlmNzk3YWVhMzA4MzI0ZmQ5NzYzNTY2NDQyNWUxZjYxZTg3YTUxNWQwYzVmNWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pFoWx5SyjDw7XU1O-8WSk13Xmux6oxbV7_nqGjwepGo"> I got sick of fixing errors. It works and passes all visual tests etc. The only major issue i have ignored is this: https://github.com/avian-annotator/FYP/blob/13c622417d33d46005124ef8f3293c0786b443b7/frontend/src/components/canvas/Canvas.tsx#L9C1-L20C1 The types are kinda complicated due to the interchangable functions. When i have more time i'll fix this tech debt but its honestly just for code quality not for funcitonality because it works.

Assuming it's from the linter, happy to chuck the canvas folder into the linter ignore, or perhaps change the severity specifically for that folder. Though generally we wouldnt do this in industry

V01DBREAKER commented:
I've fixed them/ignored them as you can see from my previous comment. Will fix in a future sprint as for now i just want functionality in and they're mostly code quality and shouldn't affect production.

V01DBREAKER committed:
fix(AAS-6): fix prettier formatting

V01DBREAKER commented:
This is a no no but ive rewritten history because my rebases were horrible. Sorry guys.

V01DBREAKER committed:
fix(AAS-6): pr-41 fix merge conflicts between develop and aas-6
